csatar-plugins/csatar/controllers/PermissionsMatrix.php

```diff
@@ -133,12 +134,32 @@ public function onSave(){
                     Log::warning($warning);
                     \Flash::warning($warning);
                 }
+
+                if ($action === 'read' && $this->checkIfStructureModelGeneralPermissionsChanged($permissionsIdsToUpdate)) {
+                    StructureTree::updateassociationTreePermissions();
+                }
             }
         }
 
         Session::forget('permissionValueChanges');
     }
 
+    public function checkIfStructureModelGeneralPermissionsChanged($permissionIds) {
+        $mandatePermissions = MandatePermission::whereIn('id', $permissionIds)
+            ->whereIn('model', [
+                '\Csatar\Csatar\Models\Association',
+                '\Csatar\Csatar\Models\District',
+                '\Csatar\Csatar\Models\Team',
+                '\Csatar\Csatar\Models\Troop',
+                '\Csatar\Csatar\Models\Patrol',
+                '\Csatar\Csatar\Models\Scout',
+            ])
+            ->where('field', MandatePermission::MODEL_GENERAL_VALUE)
+            ->get();
```

Comment:
Review this

Reply:
It is ok, this method checks if any of the permission ids belong to a model that we are interested in, and if the field type is MODEL_GENERAL.